src/coreclr/vm/codeman.cpp

```diff
@@ -1402,18 +1402,21 @@ void EEJitManager::SetCpuInfo()
         CPUCompileFlags.Set(InstructionSet_X86Serialize);
     }
 
+    if (((cpuFeatures & XArchIntrinsicConstants_Gfni) != 0) && CLRConfig::GetConfigValue(CLRConfig::EXTERNAL_EnableGFNI))
+    {
+        CPUCompileFlags.Set(InstructionSet_GFNI);
+        CPUCompileFlags.Set(InstructionSet_GFNI_V256);
+        CPUCompileFlags.Set(InstructionSet_GFNI_V512);
+    }
+
     if (((cpuFeatures & XArchIntrinsicConstants_Evex) != 0) &&
         ((cpuFeatures & XArchIntrinsicConstants_Avx10v1) != 0))
     {
         if (CLRConfig::GetConfigValue(CLRConfig::EXTERNAL_EnableAVX10v1))
         {
             CPUCompileFlags.Set(InstructionSet_EVEX);
             CPUCompileFlags.Set(InstructionSet_AVX10v1);
-
-            if((cpuFeatures & XArchIntrinsicConstants_Avx512) != 0)
```

Comment:
This check wasn't really correct because it looks only at the cpuid bit and not the config knob.  However, it doesn't matter anyway because if AVX512 is disabled, `AVX10v1_V512` will be removed by implication in the call to `EnsureValidInstructionSetSupport`.  For simplicity and consistency with the other combo flags to come, I changed it to add the whole set.